### PR TITLE
decomp: `projectiles`

### PR DESCRIPTION
**Pull request by xTVaser**
blocked by #754

### CODE REVIEW

File: goal_src/engine/game/projectiles.gc [hunk 1]
```diff
@@ -5,3 +5,3358 @@
...
+   (vector+float*!
+    (-> obj target)
+    (-> obj root-override trans)
+    (-> obj root-override transv)
+    2.0
+    )
+   (set! (-> obj target-base quad) (-> obj target quad))
+   (let
+    ((f30-0
+      (the float (sar (shl (the int (y-angle (-> obj root-override))) 48) 48))
```
Comment (water111) [inline goal_src/engine/game/projectiles.gc:2849]: all these weird shifts are a cool trick to wrap angles to +/- 180 degrees

File: goal_src/engine/game/projectiles.gc [hunk 2]
```diff
@@ -5,3 +5,3358 @@
...
+   )
+  (+! (-> obj target-count) 1)
+  0
+  (none)
+  )
+
+;; definition for method 27 of type projectile-blue
+;; INFO: Return type mismatch int vs none.
+;; Used lq/sq
+(defmethod dummy-27 projectile-blue ((obj projectile-blue))
```
Comment (water111) [inline goal_src/engine/game/projectiles.gc:3160]: this is interesting. This is code for some other projectile that plays the blue eco sound I guess?

**Review by water111 — COMMENTED** [review]
Looks good, just one comment on types.

File: goal_src/engine/game/projectiles.gc [hunk 3]
```diff
@@ -5,3 +5,3358 @@
...
+  (vector+float*!
+   (-> self target)
+   (-> self root-override trans)
+   (-> self root-override transv)
+   2.0
+   )
+  (set! (-> self target-base quad) (-> self target quad))
+  (dummy-27 self)
+  (dummy-24 self)
+  (when (not (type-type? (-> self type) projectile-blue))
```
[the remark this answers is not in the document]
Comment (xTVaser) [reply goal_src/engine/game/projectiles.gc:2807]: i _think_ they are blue eco related, but im not 100% sure.  I've found some in the game but can't remember where. shooting yellow eco is definitely an instance of `projectile-yellow` though.

**Pull request merged**